<launch>
  <!-- 
    start Nao control and sensor nodes so that they connect to a local NaoQI
    (e.g. a local simulated one or when running directly on the robot).
    
    For a remote connection, change IP and port accordingly in the NAO_IP and NAO_PORT 
    environment variables.
  -->
  
  <!-- 
  	This pushes the local PYTHONPATH into the launch file, so that the NaoQI API is found.
    You need to add the Nao's API dir to your PYTHONPATH so that the modules are found.
  -->
  <env name="PYTHONPATH" value="$(env PYTHONPATH)" /> 

  <arg name="nao_ip" default="$(optenv NAO_IP 10.0.3.16)" />
  <arg name="nao_port" default="$(optenv NAO_PORT 9559)" />
  <arg name="force_python" default="true" />

  <!-- Start nodes in their python variant (only when setting force_python:=true during launch) -->
  <group if="$(arg force_python)">
    <node pkg="nao_driver" type="nao_sensors.py" name="nao_sensors" required="true" args="--pip=$(arg nao_ip) --pport=$(arg nao_port)" />
  </group>
  
  <!-- Start nodes in their C++ variant (default, when force_python:=false) -->
  <group unless="$(arg force_python)">
    <node pkg="nao_driver" type="nao_sensors_cpp" name="nao_sensors" required="true" args="--pip=$(arg nao_ip) --pport=$(arg nao_port)" />
  </group>

  <!-- All other nodes (only exist in one varian) -->
  
  <node pkg="nao_driver" type="nao_walker.py" name="nao_walker" required="true" args="--pip=$(arg nao_ip) --pport=$(arg nao_port)" />
  <node pkg="nao_driver" type="nao_controller.py" name="nao_controller" required="true" args="--pip=$(arg nao_ip) --pport=$(arg nao_port)" />
  <node pkg="nao_driver" type="nao_behaviors.py" name="nao_behaviors" required="true" args="--pip=$(arg nao_ip) --pport=$(arg nao_port)" />
  <node pkg="nao_driver" type="nao_diagnostic_updater.py" name="nao_diagnostic_updater" args="--pip=$(arg nao_ip) --pport=$(arg nao_port)" />
  <node pkg="nao_driver" type="nao_leds.py" name="nao_leds" args="--pip=$(arg nao_ip) --pport=$(arg nao_port)" />
  <node pkg="nao_driver" type="nao_logger.py" name="nao_logger" args="--pip=$(arg nao_ip) --pport=$(arg nao_port)" />
  <node pkg="nao_driver" type="nao_speech.py" name="nao_speech" args="--pip=$(arg nao_ip) --pport=$(arg nao_port)" />
  <node pkg="nao_driver" type="nao_tactile.py" name="nao_tactile" args="--pip=$(arg nao_ip) --pport=$(arg nao_port)" />

  <!-- Load left and right sonar -->
  <include file="$(find naoqi_sensors_py)/launch/sonar.launch" ns="nao_robot/sonar/left" >
    <arg name="memory_key" value="Device/SubDeviceList/US/Left/Sensor/Value" />
    <arg name="frame_id" value="LSonar_frame" />
  </include>
  <include file="$(find naoqi_sensors_py)/launch/sonar.launch" ns="nao_robot/sonar/right" >
    <arg name="memory_key" value="Device/SubDeviceList/US/Right/Sensor/Value" />
    <arg name="frame_id" value="RSonar_frame" />
  </include>

    <!-- Specify here all dependent settings (default top) -->
  <include file="$(find naoqi_sensors_py)/launch/camera.launch" ns="nao_robot/camera/top" />
  <include file="$(find naoqi_sensors_py)/launch/camera.launch" ns="nao_robot/camera/bottom" >
    <arg name="source" value="1" />
  </include>


</launch>
